### PR TITLE
fix(timeouts): return the right socket timeout error

### PR DESCRIPTION
**Pull request by judofyr**

### CODE REVIEW

File: src/request/node-request.ts [hunk 1]
```diff
@@ -217,6 +217,7 @@ export const httpRequester: HttpRequest = (context, cb) => {
   ) as FinalizeNodeOptionsPayload
   const request = transport.request(finalOptions, (response) => {
     const res = tryCompressed ? decompressResponse(response) : response
+    ;(request as any)._getItResponse = res
```
Comment (judofyr) [inline src/request/node-request.ts:220]: Is there a better way of doing this?

Comment (stipsan) [reply src/request/node-request.ts:220]: > "Det enkle er ofte det beste" - Rema 1000

I can think of prettier solutions with `WeakMap` but then you have to do cleanup 🤷 :this-is-fine:

**Review by stipsan — DISMISSED** [review]

**Review by sgulseth — COMMENTED** [review]
Pushed an alternate approach that doens't require to set the response on the request object here:
https://github.com/sanity-io/get-it/compare/socket-timeout-fixes...alt-socket-timeout-fixes?expand=1

also improved some typings. WDYT?

File: src/request/node/timedOut.ts [hunk 2]
```diff
@@ -37,17 +36,21 @@ export function timedOut(req: any, time: any) {
...
-        req.abort()
+      socket.setTimeout(delays.socket, function socketTimeoutHandler() {
         const e: any = new Error('Socket timed out on request' + host)
         e.code = 'ESOCKETTIMEDOUT'
-        req.emit('error', e)
+        // HACK: The official documentation (https://nodejs.org/api/http.html#httprequesturl-options-callback)
+        // claims that calling `req.destroy(err)` will emit the error on the response object as well.
+        // However, I've never been able to reproduce this behavior. It always ends up being called with
+        // "Error: aborted" instead. We really want the original error to surface. We workaround this
+        // by destroying the response object _first_.
```
[the remark this answers is not in the document]
Comment (judofyr) [reply src/request/node/timedOut.ts:50]: > 'error' on the res object with an error with message 'Error: aborted' and code 'ECONNRESET', **or the error with which req.destroy() was called**

This is not what I'm seeing. I'm seeing `Error: aborted` _always_ regardless of the error I pass into `req.destroy`.

**Comment by judofyr** [comment]
@sgulseth: I think your 

> Pushed an alternate approach that doens't require to set the response on the request object here: https://github.com/sanity-io/get-it/compare/socket-timeout-fixes...alt-socket-timeout-fixes?expand=1
> 
> also improved some typings. WDYT?

Agreed! Merged it as-it since the tests passed. However, I also discovered that your change actually make us return multiple/duplicate errors on the channel (which I don't believe is what we want). I've improved the tests and pushed a fix as well.

**Review by sgulseth — APPROVED** [review]
Ah, nice! I had a case where no error got thrown, that's why I removed the check on error 🤔 but can't reproduce it now, must have been pebkac! Lgtm

**Review by stipsan — APPROVED** [review]
Nice 🥳

**Pull request merged**